Source/Core/VideoCommon/TextureCacheBase.h

```diff
@@ -170,10 +223,26 @@ class TextureCacheBase
   TCacheEntry* Load(const u32 stage);
   static void InvalidateAllBindPoints() { valid_bind_points.reset(); }
   static bool IsValidBindPoint(u32 i) { return valid_bind_points.test(i); }
-  void BindTextures();
+  TCacheEntry* GetTexture(u32 address, u32 width, u32 height, const TextureFormat texformat,
+                          const int textureCacheSafetyColorSampleSize, u32 tlutaddr = 0,
+                          TLUTFormat tlutfmt = TLUTFormat::IA8, bool use_mipmaps = false,
+                          u32 tex_levels = 1, bool from_tmem = false, u32 tmem_address_even = 0,
+                          u32 tmem_address_odd = 0);
+
+  TCacheEntry* GetXFBTexture(u32 address, u32 width, u32 height, const TextureFormat texformat,
+    const int textureCacheSafetyColorSampleSize);
+  std::optional<TextureLookupInformation> ComputeTextureInformation(u32 address, u32 width, u32 height, const TextureFormat texformat,
+    const int textureCacheSafetyColorSampleSize, bool from_tmem, u32 tmem_address_even, u32 tmem_address_odd, u32 tlutaddr, TLUTFormat tlutfmt, u32 levels);
+  TCacheEntry* GetXFBFromCache(const TextureLookupInformation& tex_info);
+  bool LoadTextureFromOverlappingTextures(TCacheEntry* entry_to_update, const TextureLookupInformation& tex_info);
+  TCacheEntry* CreateNormalTexture(const TextureLookupInformation& tex_info);
+  void LoadTextureFromMemory(TCacheEntry* entry_to_update, const TextureLookupInformation& tex_info);
+  void LoadTextureLevelZeroFromMemory(TCacheEntry* entry_to_update, const TextureLookupInformation& tex_info, bool decode_on_gpu);
+  void LoadTextureMipMapsFromMemory(TCacheEntry* entry_to_update, const TextureLookupInformation& tex_info, bool decode_on_gpu);
+  virtual void BindTextures();
```

Comment:
These virtuals should probably be grouped together somewhere. When they're spread out all over the place like this, it's a pain to see what part of the interface needs to be implemented or what can be overridden.

Reply:
I don't think I've added any virtuals, I might have moved this function accidentally.  If so, I'll move it back.

But yeah, TexturecacheBase is a mess.  Once Hybrid XFB is merged I plan on doing some cleanup work before moving to anything else.

Reply:
Hmm, I put these functions in the middle of some other ones.  But to be fair, there's virtual functions everywhere.  So this likely falls into the "clean up later" PR.